pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -118,6 +124,8 @@
   private volatile long _lastIndexedTimeMs = Long.MIN_VALUE;
   private volatile long _latestIngestionTimeMs = Long.MIN_VALUE;
 
+  private RealtimeLuceneReadersForRealtimeSegment _realtimeLuceneReadersForRealtimeSegment;
```

Comment:
Also, does it make sense to make this a final and create it right here? It is only memory, and we can add it to the refresh thead if we find that are there any columns that have lucene indexing. You can skip some if checks in the code below

Reply:
If it is lazily initialized, the code knows if this object has to be added to global concurrent queue or not.